gsa/src/gmp/models/target.js

```diff
@@ -109,13 +109,18 @@ class Target extends Model {
       delete ret.portList;
     }
 
-    for (const name of HYPERION_TARGET_CREDENTIAL_NAMES) {
-      const cred = ret[name];
-      if (hasValue(cred) && hasValue(cred.id)) {
-        ret[name] = parseModelFromObject(cred, 'credential');
-      } else {
-        delete ret[name];
+    if (hasValue(ret.credentials)) {
+      // in some instances we do not query for target credentials
+      for (const key of Object.keys(HYPERION_TARGET_CREDENTIAL_NAMES)) {
+        const cred = ret.credentials[HYPERION_TARGET_CREDENTIAL_NAMES[key]];
+        if (hasValue(cred) && hasValue(cred.id)) {
+          ret[key] = parseModelFromObject(cred, 'credential');
```

Comment:
We should use the new structure for the model as well. No need to add separate fields for the different credentials again. It should be the same structure in all components.